src/api/java/com/minecolonies/coremod/colony/requestsystem/IRequestManager.java

```diff
@@ -0,0 +1,95 @@
+package com.minecolonies.coremod.colony.requestsystem;
+
+import com.minecolonies.coremod.colony.IColony;
+import com.minecolonies.coremod.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.coremod.colony.requestsystem.location.ILocatable;
+import com.minecolonies.coremod.colony.requestsystem.request.IRequest;
+import com.minecolonies.coremod.colony.requestsystem.resolver.IRequestResolverProvider;
+import com.minecolonies.coremod.colony.requestsystem.token.IToken;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.INBTSerializable;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface used to describe classes that function as managers for requests inside a colony.
+ * Extends INBTSerializable to allow for easy reading and writing from NBT.
+ */
+public interface IRequestManager extends INBTSerializable<NBTTagCompound> {
+
+    /**
+     * The colony this manager manages the requests for.
+     * @return The colony this manager manages the requests for.
+     */
+    @NotNull
+    IColony getColony();
+
+    /**
+     * Method used to get the FactoryController of the RequestManager.
+     * @return The FactoryController of this RequestManager.
+     */
+    @NotNull
+    IFactoryController getFactoryController();
+
+    /**
+     * Method to create a request for a given object
+     * @param requester The requester.
+     * @param object The Object that is being requested.
+     * @param <T> The type of request.
+     * @return The token representing the request.
+     * @throws IllegalArgumentException is thrown when this manager cannot produce a request for the given types.
```

Comment:
I wouldn't make it a null value. I would make it really a Token.Invalid with a concrete value.
We had a lot of exceptions in minecolonies which we removed almost completely because they did more bad than good.